contracts/WildcardSteward_v2.sol

```diff
@@ -466,17 +530,110 @@ contract WildcardSteward_v2 is Initializable {
         address currentOwner = assetToken.ownerOf(tokenId);
         address tokenPatron = currentPatron[tokenId];
 
-        if (state[tokenId] == StewardState.Owned) {
-            uint256 totalToPayBack = price[tokenId];
-            // NOTE: pay back the deposit only if it is the only token the patron owns.
-            if (
-                totalPatronOwnedTokenCost[tokenPatron] ==
-                price[tokenId].mul(patronageNumerator[tokenId])
-            ) {
-                totalToPayBack = totalToPayBack.add(deposit[tokenPatron]);
-                deposit[tokenPatron] = 0;
-            }
+        _buy(tokenId, tokenPatron, wildcardsPercentage);
+        _handleBuy(
+            tokenId,
+            wildcardsPercentage,
+            remainingValueForDeposit,
+            currentOwner,
+            tokenPatron,
+            _newPrice
+        );
+    }
+
+    // Check possible issues with collectPatronage(tokenId) [probably leave out since no patronage to collect on forecolsed token!?]
+    function buyAuction(
+        uint256 tokenId,
+        uint256 _newPrice,
+        uint256 wildcardsPercentage
+    )
+        public
+        payable
+        collectPatronage(tokenId)
+        collectPatronageAddress(msg.sender)
+        priceGreaterThanZero(_newPrice)
+        validWildcardsPercentage(wildcardsPercentage)
+    {
+        require(
+            state[tokenId] == StewardState.Foreclosed,
+            "Can only buy foreclosed tokens useing this function"
+        );
+        // Calculate auction price for the token and replace price[tokenId] below
+        uint256 auctionTokenPrice = _auctionPrice(tokenId);
+
+        // Note: We need to calculate the requisite value on our front end using same function
+        uint256 remainingValueForDeposit = msg.value.sub(auctionTokenPrice);
```

Comment:
I wonder if this would benefit from the same system where a minimum ValueForDeposit is checked, since the operator could front-run the TX to set the fee really high with changeAuctionParamters

Reply:
That is a fair point, but I think given we are the only ones who can change the auction parameters (we are the operator) this isn't a significant issue.

The buy (if the token is owned by another user) and auction function are separate on purpose to prevent some issues where the parameters change due to someone buying the auction before you. It is not perfect, but frontrunning in the auction isn't a huge concern IMO.